compiler/next/include/chpl/uast/For.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_FOR_H
+#define CHPL_UAST_FOR_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/IndexableLoop.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a for loop. For example:
+
+  \rst
+  .. code-block:: chapel
+
+      // Example 1:
+      for i in foo {
```

Comment:
I'd expect `foo` to have parens here.

Reply:
What about something that has a `these`?

Reply:
Of course. To me `foo` is more a function than a variable. If we prefer going without parens, I suggest using something like `myRange` instead of `foo`.

Reply:
Sounds good, thanks!